src/pages/Order.tsx

```diff
@@ -535,12 +706,90 @@ function Order() {
 
       {/* 주문 버튼 */}
       <OrderBtnWrapper>
-        <OrderButton onClick={() => onClickHandler()}>
+        <OrderButton onClick={handleSubmit(handleOrderClick)}>
           {price * itemCount}원 주문하기
         </OrderButton>
       </OrderBtnWrapper>
+
+      {/* --------------모달-------------- */}
+      {modalToggle && <ModalOverlay>
+        <ModalContent>
+          <ModalInfoWrapper>
+            <ModalTitle>받는 사람</ModalTitle>
+            <ModalInfoTxt>* 최대 10명까지 추가 할 수 있어요.</ModalInfoTxt>
+            <ModalInfoTxt>* 받는 사람의 전화번호를 중복으로 입력할 수 없어요.</ModalInfoTxt>
+          </ModalInfoWrapper>
+          <ModalInfoAddBtn onClick={handleReceiverAdd} disabled={fields.length >= 10}>추가하기</ModalInfoAddBtn>
+
+          {/* 모달 리시버 박스들 */}
+          {fields.map((field, index) => (
+            <ModalReceiverInputBox key={field.id}>
+            <ModalReceiverInputTitleBtnWrapper>
+              <ModalReceiverInputTitle>받는 사람 {index + 1}</ModalReceiverInputTitle>
+              <ModalReceiverInputDelBtn onClick={() => handleReceiverDel(index)}>
+                <svg xmlns="http://www.w3.org/2000/svg" width="20" height="20" viewBox="0 0 24 24"
+                  fill="none" stroke="currentColor" strokeWidth="1.5" strokeLinecap="round"
+                  strokeLinejoin="round" className="lucide lucide-x" aria-hidden="true">
+                  <path d="M18 6 6 18"></path>
+                  <path d="m6 6 12 12"></path>
+                </svg>
+              </ModalReceiverInputDelBtn>
+            </ModalReceiverInputTitleBtnWrapper>
+      
+            <ReceiverInputNameWrapper>
+              <ReceiverInputNameLabel htmlFor={`receivers.${index}.name`}>이름</ReceiverInputNameLabel>
+              <ReceiverInputName
+                placeholder="이름을 입력하세요."
+                {...register(`receivers.${index}.name`, { required: true})}
+              />
+            </ReceiverInputNameWrapper>
+            {errors.receivers?.[index]?.name && <ReceiverInputErrorTxt>이름을 입력해 주세요.</ReceiverInputErrorTxt>}
+      
+            <ReceiverInputPhoneNumberWrapper>
+              <ReceiverInputPhoneNumberLabel htmlFor={`receivers.${index}.phone`}>전화번호</ReceiverInputPhoneNumberLabel>
+              <ReceiverInputPhoneNumber
+                placeholder="전화번호를 입력하세요"
+                {...register(`receivers.${index}.phone`, {
+                  required: true,
+                  pattern: {
+                    value: PHONE_NUM_REGEX,
+                    message: '전화번호 형식이 맞지 않습니다.'
+                  }
```

Comment:
> 중복된 전화번호 검증은 너무어려워서 못했습니다...

register에서 validate 속성을 통해 전화번호 중복 검사와 메세지를 할당하는 방식으로 구현하실 수 있을거에요 💪

1. 포넌트 상단에서 현재 입력된 전화번호 목록을 가져오고
```ts
const receivers = watch('receivers');
const phoneNumbers = receivers.map((receiver) => ...);
```

2. register validate에서 받는 사람 필드 상태들에서 중복된 전화번호가 있는지 검사

```suggestion
                  required: true,
                  pattern: {
                    value: PHONE_NUM_REGEX,
                    message: '전화번호 형식이 맞지 않습니다.'
                  }
                  validate: (value) => {
                    const isDuplicated = 전화번호목록 중복 있는지 검사...
                    
                    if (isDuplicated) {
                      return '이미 등록된 전화번호입니다.';
                    }
                    
                    return true;
                   },
```